test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommand_PackagesLockFileTests.cs

```diff
@@ -0,0 +1,951 @@
+                var intermediateProjectReferenceSpec = PackageReferenceSpecBuilder.Create(projectName, projectDirectory)
+                    .WithTargetFrameworks(new string[] { targetFramework.GetShortFolderName() })
+                    .Build();
+                allPackageSpecs.Add(intermediateProjectReferenceSpec);
+
+
+                projectName = "LeafProject";
+                projectDirectory = Path.Combine(pathContext.SolutionRoot, projectName);
+                var leafProjectReferenceSpec = PackageReferenceSpecBuilder.Create(projectName, projectDirectory)
+                    .WithTargetFrameworks(new string[] { targetFramework.GetShortFolderName() })
+                    .Build();
+                allPackageSpecs.Add(leafProjectReferenceSpec);
+
+                // Add the dependency to all frameworks
+                PackageSpecOperations.AddOrUpdateDependency(rootPackageSpec, packageA.Identity, rootPackageSpec.TargetFrameworks.Select(e => e.FrameworkName));
+                PackageSpecOperations.AddOrUpdateDependency(intermediateProjectReferenceSpec, packageB.Identity, rootPackageSpec.TargetFrameworks.Select(e => e.FrameworkName));
+                PackageSpecOperationsUtility.AddProjectReference(rootPackageSpec, intermediateProjectReferenceSpec, targetFramework);
+                PackageSpecOperationsUtility.AddProjectReference(intermediateProjectReferenceSpec, leafProjectReferenceSpec, targetFramework);
+
+                // Preconditions.
+                var result = await new RestoreCommand(ProjectTestHelpers.CreateRestoreRequest(rootPackageSpec, allPackageSpecs, pathContext, logger)).ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+                result.Success.Should().BeTrue();
+
+                // Enable locked mode, remove a dependency and clear the logger;
+                rootPackageSpec.RestoreMetadata.RestoreLockProperties = new RestoreLockProperties(
+                    restorePackagesWithLockFile: "true",
+                    rootPackageSpec.RestoreMetadata.RestoreLockProperties.NuGetLockFilePath,
+                    restoreLockedMode: true);
+                logger.Clear();
+
+                PackageSpecOperations.AddOrUpdateDependency(intermediateProjectReferenceSpec, packageC.Identity, rootPackageSpec.TargetFrameworks.Select(e => e.FrameworkName));
+
+                // Act.
+                result = await new RestoreCommand(ProjectTestHelpers.CreateRestoreRequest(rootPackageSpec, allPackageSpecs, pathContext, logger)).ExecuteAsync();
+
+                // Assert.
+                result.Success.Should().BeFalse();
+                logger.ErrorMessages.Count.Should().Be(1);
+                logger.ErrorMessages.Single().Should().Contain("NU1004");
+                logger.ErrorMessages.Single().Should().Contain($"The project reference intermediateproject has changed. Current dependencies count: 3, lock file's dependencies count: 2.");
```

Comment:
same as the above.

Reply:
Updated the message to list the dependencies instead of counts.